doto-backend/API_documentation.md

```diff
@@ -0,0 +1,305 @@
+# Register
+
+Used to register the user for the application
+
+**URL** : `/api/schedule/post`
+
+**Method** : POST
+
+** Auth required** : NO
+
+**Data constraints**
+
+```json
+{
+	"Username" : "[Unique email address]",
+	"Password" : "[Password the user chooses]"
+}
+```
+**Data example**:
+```json
+{
+	"Username" : "jyao413@aucklanduni.ac.nz",
+	"Password"  : "softwareengineering"	
+}
+```
+## Success response
+
+**Code** : `200 OK`
+
+**Content examples**
+
+For user that wants to register their account without doing it before
+```json
+{
+    "_id": "1234",
+    "Successful": "True"
+}
+```
+For user that wants to register their account again
+```json
+{
+    "_id": " ",
+    "Successful": "False"
+}
+```
+
+# Notes: 
+
+User should not be able to register twice, and the username (email) of the account must be unique
+
+# Login
+
+Used to log the user into the application
+
+**URL** : `/api/login`
+
+**Method** : `POST`
+
+**Auth required** : YES
+
+**Data constraints**:
+```json
+{
+	"Username" : "[Unique email address]",
+	"Password"  : "[Password the user chooses]"
+}
+```
+**Data example**:
+```json
+{
+	"Username" : "jyao413@aucklanduni.ac.nz",
+	"Password"  : "softwareengineering"	
+}
+```
+## Success response
+
+**Code** : 200 OK
+
+**Content examples**
+
+For a user with ID 1234 on the database where the user has saved their login information:
+```json
+{
+    "name": "Jason",
+    "email": "jyao413@aucklanduni.ac.nz",
+    "_id": "1234", 
+        "Picture": "imgur.com/ao12r",
+        "Preferences": "blue",
+    "__v": "0"
+}
+```
+Notes: 
+
+User first has to register
+
+
+# Schedule a task
+
+Used to schedule a task
+
+**URL** : `/api/schedule/post`
+
+**Method** : `POST`
+
+**Auth required** : NO
+
+**Data constraints**:
+```json
+{
+	"id" : "Unique id for task]",
+	"Title" : "Name for schedule task",
+	"Description"  : "[Description for task]",
+	"Start_Date" : "[Time of creation]",
+	"End_Date" : "[Time to finish task by]",
+	"User_id" : "[Users email]",
```

Comment:
hmm, i'm not a huge fan of using the users email as the id... i would much prefer if it was the user id that was assigned when creating the account. if we do this we need a /get request that returns user information such as id. or i guess we could just store id in their session

Reply:
I think we'll be using the google unique id going forward.